src/main/java/org/spoken_tutorial/health/elasticsearch/JsonService/JsonService.java

```diff
@@ -83,4 +84,165 @@ public String saveNarrationToFile(String url, String documentId) throws ParseExc
         }
         return document;
     }
+
+    public String saveNarrationAndCuefScriptoHtmlFile(String url, int tutorialId, int lanId)
+            throws ParseException, IOException {
+
+        String jsonString = "";
+        boolean flag = true;
+        jsonString = restTemplate.getForObject(url, String.class);
+
+        String document = "";
+
+        Path jsonDir = Paths.get(mediaRoot, Config.uploadDirectoryScriptJsonFile);
+        Files.createDirectories(jsonDir);
+        Path jsonfilePath = Paths.get(mediaRoot, Config.uploadDirectoryScriptJsonFile, tutorialId + ".json");
+        File jsonfile = jsonfilePath.toFile();
+        if (jsonfile.exists()) {
+            jsonString = new String(Files.readAllBytes(Paths.get(jsonfilePath.toString())));
+
+        } else {
+            Files.writeString(jsonfilePath, jsonString);
```

Comment:
jsonString = restTemplate.getForObject(url, String.class);

Move above line before this line.

Reply:
it cant be moved , because we are using if and else.

Reply:
if already has jsonString being defined, it should not be required there.